Sources/Overture/Chain.swift

```diff
@@ -90,6 +90,98 @@ public func chain<A, B, C, D, E, F, G>(
     }
 }
 
+
+/// Forward composition of functions that return optionals and throw. Useful for chaining operations that may fail.
+///
+///     chain(String.init(validatingUTF8:), URL.init(string:))
```

Comment:
Let's try to update this example with a throwing function, or let's just get rid of it for now.